specs/vep-2272-complete-data-job-configuration-persistence/README.md

```diff
@@ -0,0 +1,161 @@
+- [Motivation](#motivation)
+- [Requirements and goals](#requirements-and-goals)
+- [High-level design](#high-level-design)
+- [API Design](#api-design)
+- [Detailed design](#detailed-design)
+- [Implementation stories](#implementation-stories)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Currently, the deployment configuration of data jobs is partially dependent on the Kubernetes cluster, leading to
+unnecessary overhead when loading data job deployments. Additionally, the loss of Kubernetes namespace results in the
+potential loss of data jobs, requiring a manual and complex restoration process. By switching the source of truth from
+Kubernetes to a database, we can ensure a more reliable and easy restoration process while optimizing performance for
+Control Service APIs.
+
+## Glossary
+
+* VDK: https://github.com/vmware/versatile-data-kit/wiki/dictionary#vdk
+* Control Service: https://github.com/vmware/versatile-data-kit/wiki/dictionary#control-service
+* Data Job: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job
+* Data Job Deployment: https://github.com/vmware/versatile-data-kit/wiki/dictionary#data-job-deployment
+* Kubernetes: https://kubernetes.io/
+
+## Motivation
+
+Ensuring efficient and reliable management of data job deployments is crucial for the smooth functioning of our system.
+Currently, when a data job is deployed, its deployment configuration is stored in both Kubernetes and the database.
+However, there exists a disparity where certain essential properties are exclusively stored in Kubernetes, such as the
+job's enabled status, Python version, and deployment information.
+
+This discrepancy creates a significant challenge for the Control Service APIs that rely on retrieving data job
+deployments. For instance, the GraphQL API's jobs query initially retrieves the data jobs from the database and
+subsequently retrieves the deployment data from Kubernetes. This two-step process introduces performance degradation to
+the Control Service, as it requires additional time and resources to fetch information from multiple sources.
+
+Moreover, in the event of a Kubernetes namespace loss, there is a potential risk of losing critical data job deployment
+configurations. In such scenarios, recovering the lost data jobs becomes a manual and complex restoration process.
+This adds further complexity and potential downtime for users relying on the Control Service for their data job
+management needs.
+
+To address these issues, it is essential to optimize the storage and retrieval of data job deployments. By ensuring that
+all relevant properties are stored consistently, in the database, so we can streamline the process and enhance the
+overall the efficiency of the Control Service APIs.
+
+By improving the efficiency and reliability of data job deployments, we aim to optimize the overall system performance
+and provide a seamless experience for our users.
+
+## Requirements and goals
+
+### Goals
```

Comment:
It is good idea to specify the metrics you will use to evaluate the success of this feature/change, such as how much performance improvement in the Control Service APIs we expect or what specific improvements in the restoration process we aim to achieve . It is ok if they are pretty rough  at this stage.

Reply:
Based on the very basic performance measurement we will gain about 35% better response times.

Test includes:
Two GraphQL queries with and without deployments. The one with deployments takes on average about 600 ms whereas the one without deployments takes on average about 400 ms.

I will incorporate that.